server/revision/revision.controller.js

```diff
@@ -15,14 +17,42 @@ function index({ course, query }, res) {
   return Revision.findAll(opts).then(data => res.json({ data }));
 }
 
+function restore({ body: { revision }, user }, res) {
+  const { id, state } = revision;
+  const opts = { recursive: true, isRestore: true, context: { userId: user.id } };
+  const include = [{ model: User, attributes: ['id', 'email'] }];
+  Revision.update({ restored: true }, { where: { id } });
+  if (revision.entity === 'ACTIVITY') return restoreActivity(state.id, opts, include, res);
```

Comment:
Why didn't you destructure ```entity``` from revision like you did ```id & state```? Why ```revision``` is property of body, why didn't you just send revision properties that you need in body?